rest/admin_api.go

```diff
@@ -260,17 +260,27 @@ func readReplicationParametersFromJSON(jsonData []byte) (params sgreplicate.Repl
 				err = base.HTTPErrorf(http.StatusBadRequest, "/_replicate sync_gateway/bychannel filter; Missing query_params")
 				return
 			}
-			var channels []string
-			err := json.Unmarshal([]byte(in.QueryParams), &channels)
-			if err != nil {
+
+			//The Channels must be passed as an array of strings
+			if chanarray, ok := in.QueryParams.([]interface{}); ok {
```

Comment:
I was thinking of a cast straight from {}interface to []string.  And wouldn't we want to throw an error on mixed type arrays anyway?